### ISSUE TITLE
Docker Using Proxy although it is disabled in Network Settings

### ISSUE THREAD

**Issue by vikas027** [docker/for-mac#266]
### Expected behavior

Docker should not pick the proxies unless specified or enabled
### Actual behavior

Docker is automatically picking up proxy even though it is disabled
![2016-08-14_02-52-07](https://cloud.githubusercontent.com/assets/6998650/17644443/34bbc5ea-61ca-11e6-8d9d-c10240b38f2a.png)

![2016-08-14_02-55-14](https://cloud.githubusercontent.com/assets/6998650/17644460/883e156a-61ca-11e6-993f-3ce525818a95.png)

I used this proxy 4-5 months ago, had a hard time to figure out where this IP 10.114.44.12 is coming from :)

``` sh
root@7ad4bddfbb31:/# apt-get update
0% [Connecting to 10.114.44.12 (10.114.44.12)]^C
root@7ad4bddfbb31:/#
```

NOTE: The issue started only after updating the recent version of Docker.
### Workaround

Remove the Proxy IP and Port completely and restart Docker
### Information

Diagnostic ID: BC3919A1-2726-462E-8CD1-2B2A2AB95329
Docker for Mac: 1.12.0-a (Build 11213)
macOS: Version 10.11.2 (Build 15C50)
[OK]     docker-cli
[OK]     app
[OK]     moby-syslog
[OK]     disk
[OK]     virtualization
[OK]     system
[OK]     menubar
[OK]     osxfs
[OK]     db
[OK]     slirp
[OK]     moby-console
[OK]     logs
[OK]     vmnetd
[OK]     env
[OK]     moby
[OK]     driver.amd64-linux
### Steps to reproduce
1. Enter a random IP and Port in Proxy
2. Restart Docker

**Comment by thaJeztah** [comment]
ping @dave-tucker

**Comment by ijc25** [comment]
@vikas027 Thanks you for the report and sorry for the delay. I believe this was fixed in the 1.12.3 stable update. Please reopen if this is not the case for you.

**Comment by hasselg** [comment]
I'm running version 1.12.3 and am also experiencing this error. Perhaps of note is that I'm using the locations feature for networks in System Preferences, and one of the locations *does* have proxy settings. The problem is that I'm seeing this error when using a location and associated interfaces that does *not* have any proxies enabled.
```
$ docker version
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Wed Oct 26 23:26:11 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   6b644ec
 Built:        Wed Oct 26 23:26:11 2016
 OS/Arch:      linux/amd64
```

When I attempt to manually edit the proxy settings for Docker via the Docker Preferences pane I see that it is pre-populating with the incorrect proxy settings. I am working around this by specifying an Exclude value of "*".
